integration-tests/src/test/java/tachyon/master/JournalCrashIntegrationTest.java

```diff
@@ -0,0 +1,172 @@
+
+  private ClientThread mCreateFileThread = null;
+  private ClientThread mCreateTableThread = null;
+  private final ExecutorService mExecutorsForClient = Executors.newFixedThreadPool(2);
+  private final ExecutorService mExecutorsForMaster = Executors.newFixedThreadPool(2);
+  private TachyonConf mMasterTachyonConf =  null;
+
+  @After
+  public final void after() throws Exception {
+    mExecutorsForClient.shutdown();
+    mExecutorsForMaster.shutdown();
+    System.clearProperty("fs.hdfs.impl.disable.cache");
+  }
+
+  @Before
+  public final void before() throws Exception {
+    System.setProperty("fs.hdfs.impl.disable.cache", "true");
+  }
+
+  /**
+   * Check if the correct state can be reproduced.
+   */
+  private void crashTestUtil(int successFiles, int successTables) throws IOException,
+      InvalidPathException, FileDoesNotExistException, TableDoesNotExistException {
+    String masterJournal = mMasterTachyonConf.get(Constants.MASTER_JOURNAL_FOLDER);
+    Journal journal = new Journal(masterJournal, "image.data", "log.data", mMasterTachyonConf);
+    MasterInfo info = new MasterInfo(new InetSocketAddress(9999), journal, mExecutorsForMaster,
+        mMasterTachyonConf);
+    info.init();
+
+    Assert.assertEquals(successFiles, info.listFiles(new TachyonURI(TEST_FILE_DIR), false).size());
+    for (int f = 0; f < successFiles; f ++) {
+      Assert.assertTrue(info.getFileId(new TachyonURI(TEST_FILE_DIR + f)) != -1);
+    }
+    Assert.assertEquals(successTables,
+        info.listFiles(new TachyonURI(TEST_TABLE_DIR), false).size());
+    for (int t = 0; t < successTables; t ++) {
+      Assert.assertTrue(info.getRawTableId(new TachyonURI(TEST_TABLE_DIR + t)) != -1);
+    }
+    info.stop();
+  }
+
+  private LocalTachyonClusterMultiMaster setupMultiMasterCluster() throws IOException {
+    LocalTachyonClusterMultiMaster ret =
+        new LocalTachyonClusterMultiMaster(100, TEST_MASTERS, TEST_BLOCK_SIZE);
+    ret.start();
+    mMasterTachyonConf = ret.getMasterTachyonConf();
+    mCreateFileThread = new ClientThread(0, ret.getClient());
+    mCreateTableThread = new ClientThread(1, ret.getClient());
+    mExecutorsForClient.submit(mCreateFileThread);
+    mExecutorsForClient.submit(mCreateTableThread);
+    return ret;
+  }
+
+  private LocalTachyonCluster setupSingleMasterCluster() throws IOException {
+    LocalTachyonCluster ret = new LocalTachyonCluster(100, 100, TEST_BLOCK_SIZE);
+    ret.start();
+    mMasterTachyonConf = ret.getMasterTachyonConf();
+    mCreateFileThread = new ClientThread(0, ret.getClient());
+    mCreateTableThread = new ClientThread(1, ret.getClient());
+    mExecutorsForClient.submit(mCreateFileThread);
+    mExecutorsForClient.submit(mCreateTableThread);
+    return ret;
+  }
+
+  @Test
+  public void singleMasterJournalCrashIntegrationTest() throws Exception {
+    LocalTachyonCluster cluster = setupSingleMasterCluster();
+    CommonUtils.sleepMs(null, TEST_TIME_MS);
+    cluster.stopTFS();
```

Comment:
I think stopTFS() is a "nice" way to stop the cluster. Is that not true? If this is a nice way to shutdown, then this test should not include "crash" in the name.

If we want true crash testing, maybe the master should really be killed at any point. Is this possible to do? Is there a way to forcibly kill all threads associated with the master?